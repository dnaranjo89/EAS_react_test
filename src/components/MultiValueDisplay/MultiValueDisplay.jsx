import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Chip from '@material-ui/core/Chip';
import Typography from '@material-ui/core/Typography';

import STYLES from './MultiValueDisplay.scss';

const c = className => STYLES[className];

class MultiValueDisplay extends Component {
  onValueDelete = value => () => this.props.onDelete(value);

  render() {
    const { values, label, messageEmpty, onDelete } = this.props;
    return (
      <div className={c('MultiValueDisplay')} data-component="MultiValueDisplay">
        {label && <Typography variant="caption">{label}</Typography>}
        <div className={c('MultiValueDisplay__values-list')}>
          {values.length ? (
            values.map(value => (
              <Chip
                key={Math.random()}
                label={value}
                onDelete={onDelete ? this.onValueDelete(value) : null}
                data-component={'MultiValueDisplay__chip'}
              />
            ))
          ) : (
            <Typography variant="body1" className={c('MultiValueDisplay__empty-message')}>
              {messageEmpty}
            </Typography>
          )}
        </div>
      </div>
    );
  }
}

MultiValueDisplay.propTypes = {
  label: PropTypes.string,
  values: PropTypes.arrayOf(PropTypes.string),
  messageEmpty: PropTypes.string,
  onDelete: PropTypes.func,
};

MultiValueDisplay.defaultProps = {
  values: [],
  label: null,
  messageEmpty: '',
  onDelete: null,
};

export default MultiValueDisplay;
