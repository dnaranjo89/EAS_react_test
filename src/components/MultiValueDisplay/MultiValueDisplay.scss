$one-pixel-rem: .0625rem;
$spacing-md: 1.125rem;
$spacing-lg: 1.875rem;
$border-radius: .2rem;
$border-color-gray: #797979;

.MultiValueDisplay {
  margin-top: $spacing-md;
  padding: $spacing-md;
  border-width: $one-pixel-rem;
  border-style: solid;
  border-radius: $border-radius;
  border-color: $border-color-gray;
}
