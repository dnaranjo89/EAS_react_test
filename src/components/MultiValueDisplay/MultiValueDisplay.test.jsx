import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MultiValueDisplay from './MultiValueDisplay.jsx';

describe('MultiValueDisplay', () => {
  beforeEach(() => {
    Math.random = jest.fn(() => 1);
  });

  it('Should render correctly without values', () => {
    const wrapper = shallow(
      <MultiValueDisplay label="Input label" name="coolInput" messageEmpty="No items selected" />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should render correctly with values', () => {
    const wrapper = shallow(
      <MultiValueDisplay
        label="Input label"
        name="coolInput"
        messageEmpty="No items selected"
        values={['value 1', 'value 2']}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Chip.onDelete should be null if onDelete is not supplied', () => {
    const wrapper = shallow(
      <MultiValueDisplay
        label="Input label"
        name="coolInput"
        messageEmpty="No items selected"
        values={['value 2']}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Chip.onDelete should be onDelete when supplied', () => {
    const wrapper = shallow(
      <MultiValueDisplay
        label="Input label"
        name="coolInput"
        messageEmpty="No items selected"
        values={['value 2']}
        onDelete={() => {}}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Deleting an item should call onDelete prop', () => {
    const onDeleteMock = jest.fn();
    const wrapper = mount(
      <MultiValueDisplay
        label="Input label"
        name="coolInput"
        messageEmpty="No items selected"
        values={['value 6']}
        onDelete={onDeleteMock}
      />,
    );
    wrapper.find('svg').simulate('click');
    expect(onDeleteMock).toHaveBeenCalledWith('value 6');
  });
});
