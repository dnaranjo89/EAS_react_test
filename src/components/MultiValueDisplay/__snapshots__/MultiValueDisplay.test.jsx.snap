// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiValueDisplay Chip.onDelete should be null if onDelete is not supplied 1`] = `
<div
  className="MultiValueDisplay"
  data-component="MultiValueDisplay"
>
  <WithStyles(Typography)
    variant="caption"
  >
    Input label
  </WithStyles(Typography)>
  <div
    className="MultiValueDisplay__values-list"
  >
    <WithStyles(Chip)
      data-component="MultiValueDisplay__chip"
      key="1"
      label="value 2"
      onDelete={null}
    />
  </div>
</div>
`;

exports[`MultiValueDisplay Chip.onDelete should be onDelete when supplied 1`] = `
<div
  className="MultiValueDisplay"
  data-component="MultiValueDisplay"
>
  <WithStyles(Typography)
    variant="caption"
  >
    Input label
  </WithStyles(Typography)>
  <div
    className="MultiValueDisplay__values-list"
  >
    <WithStyles(Chip)
      data-component="MultiValueDisplay__chip"
      key="1"
      label="value 2"
      onDelete={[Function]}
    />
  </div>
</div>
`;

exports[`MultiValueDisplay Should render correctly with values 1`] = `
<div
  className="MultiValueDisplay"
  data-component="MultiValueDisplay"
>
  <WithStyles(Typography)
    variant="caption"
  >
    Input label
  </WithStyles(Typography)>
  <div
    className="MultiValueDisplay__values-list"
  >
    <WithStyles(Chip)
      data-component="MultiValueDisplay__chip"
      key="1"
      label="value 1"
      onDelete={null}
    />
    <WithStyles(Chip)
      data-component="MultiValueDisplay__chip"
      key="1"
      label="value 2"
      onDelete={null}
    />
  </div>
</div>
`;

exports[`MultiValueDisplay Should render correctly without values 1`] = `
<div
  className="MultiValueDisplay"
  data-component="MultiValueDisplay"
>
  <WithStyles(Typography)
    variant="caption"
  >
    Input label
  </WithStyles(Typography)>
  <div
    className="MultiValueDisplay__values-list"
  >
    <WithStyles(Typography)
      className="MultiValueDisplay__empty-message"
      variant="body1"
    >
      No items selected
    </WithStyles(Typography)>
  </div>
</div>
`;
